SRS Document.md

```diff
@@ -0,0 +1,112 @@
+| LTS | stands for "Long Term Support" and refers to software that is supported for a longer length of time than the standard edition |
+|RESTful| a web API that obeys the REST constraints is informally described as RESTful |
+|Sync mode| a mode where an operation waits for a response out the server in the format of an output parameter|
+|CI tools| software applications that assist the user in efficient integration and code development |
+
+
+# 2. Overall Description
+
+## 2.1 Product Perspective
+* **JavaScript**: With the support of the WebdriverIO library, the user builds the script.
+* **NodeJS**: NodeJS is an open-source project that makes the JavaScript runtime environment easier to use.
+* **WebdriverIO**: WebdriverIO is a NodeJS-based application that interacts with NodeJS.
+
+![](https://github.com/TemirlanAidarov/Images/blob/main/Diagram.drawio.svg)
+
+
+WebdriverIO is based on NodeJS, which is considered to be a JSON Wire Protocol implementation. It performs automated testing using a RESTful architecture.
+
+With the support of the WebdriverIO library, the user builds the script in JavaScript. In that case, the Service request is transmitted through NodeJS as an HTTP instruction. It appllies the JSON Wire protocol and the service module redirects the request to the webbrowser.
+
+Then the browser executes the user actions after getting the instruction, ensuring that the application functionalities are legitimate.
+
+## 2.2 Product Functions
+Using this WebdriverIO, a user can automate modern web written in  React, Angular, Polymeror Vue.js as well as native mobile applications for Android and iOS.
+
+* WebDriver Protocol (for cross browser testing)
+* Chrome Dev Tools Protocol (for chromium based automation)
+* Appium (for mobile automation)
+
+## 2.3 User Classes and Characteristics
+In general, WebdriverIO is characterized as a simple and accessible tool for creating web and mobile app tests. It is accessible to anyone who wants to learn and write frontend tests in JavaScript (TypeScript) and needs some assistance.
+
+WebdriverIO is easy to start and get used to for beginners. The installation process is straightforward and quick. Users do not need to install any browser drivers and configure themselves. After installation, the user only needs to write a config command that will perform all the necessary configuration. Further details are described in the [documentation page](https://webdriver.io/docs/gettingstarted) 
+
+## 2.4 Operating Environment
+A user has to install Node.js. It is suggested to choose the *LTS* version, so it is suggested to choose 12.16.1 or above. 
+There are no other strict rules regarding system requirements. Nevertheless, in order to work with little problem 
+as possible with Node.js and WebdriverIO, it is advised for users to have:
+* Ubuntu version 16.04 or higher for Linux 
+* or Windows 7 or newer versions with at least 2GB RAM (4GB preferable). 
```

Comment:
I don't see how systems not listed here (e.g. Mac) should have more problems to run WebdriverIO. We test this software on all broader operating systems.